internal/cmd/start.go

```diff
@@ -4,39 +4,34 @@ import (
 	"fmt"
 	"log"
 
-	"github.com/urfave/cli"
+	"github.com/spf13/cobra"
 	"github.com/xUnholy/go-proxy/pkg/execute"
 
 	"github.com/xUnholy/go-proxy/internal/cntlm"
 	"github.com/xUnholy/go-proxy/internal/profile"
 )
 
-func StartCommand() cli.Command {
-	return cli.Command{
-		Name:        "start",
-		Aliases:     []string{""},
-		Usage:       "proxy start",
-		Description: "Start CNTLM Proxy",
-		Flags: []cli.Flag{
-			cli.IntFlag{
-				Name:        "port, p",
-				Value:       3128,
-				Usage:       "set custom CNTLM `PORT`",
-				Destination: &port,
-			},
-		},
-		Action: func(_ *cli.Context) {
-			proxyURL := makeProxyURL(port)
-			profile.UpdateGlobalEnvironmentVariables(proxyURL)
-			p := fmt.Sprintf("Listen\t%v", port)
-			cntlm.UpdateFile(cntlmFile, p)
-			cmds := execute.Command{Cmd: "cntlm", Args: []string{"-g"}}
-			_, err := execute.RunCommand(cmds)
-			if err != nil {
-				fmt.Println("CNTLM Proxy couldn't be started. Is it already running?")
-				log.Fatal(err)
-			}
-			fmt.Println("CNTLM Proxy Started On", proxyURL)
-		},
+func SetupStartCli() *cobra.Command {
+	// Top-level command
+	var startCmd = &cobra.Command{
+		Use:   "proxy start",
+		Short: "Start CNTLM Proxy",
+		Run:   startCmd,
 	}
+	startCmd.Flags().IntVar(&port, "port", 3128, "set custom CNTLM `PORT`")
+	return startCmd
+}
+
+func startCmd(cmd *cobra.Command, args []string) {
+	proxyURL := makeProxyURL(port)
+	profile.UpdateGlobalEnvironmentVariables(proxyURL)
+	p := fmt.Sprintf("Listen\t%v", port)
+	cntlm.UpdateFile(cntlmFile, p)
+	cmds := execute.Command{Cmd: "cntlm", Args: []string{"-g"}}
+	_, err := execute.RunCommand(cmds)
+	if err != nil {
+		fmt.Println("CNTLM Proxy couldn't be started. Is it already running?")
```

Comment:
We can combine the Println and log.fatal 

`log.Fatalf("CNTLM Proxy couldn't be started. Is it already running?", err)`

Reply:
Or with %q however, i think that's only used in fmt?

Reply:
Yes that can be done!

This pull request intends to *not* change any functionality.. Just a refactor

Reply:
Yeah, I'll fix it once merged - just highlighting it for either of us to do later.